src/main/java/org/uikit/VideoJNI.kt

```diff
@@ -79,15 +82,21 @@ class AVPlayer(parent: SDLActivity, asset: AVURLAsset) {
             }
 
             override fun onPlayerError(error: ExoPlaybackException) {
-                Log.e("SDL", "ExoPlaybackException occurred")
-                val message = error.message ?: "unknown"
-                nativeOnVideoError(error.type, message)
+                this@AVPlayer.userContext?.let { userContext ->
+                    Log.e("SDL", "ExoPlaybackException occurred")
+                    val message = error.message ?: "unknown"
+                    nativeOnVideoError(error.type, message, userContext)
+                }
             }
         }
 
         exoPlayer.addListener(listener)
     }
 
+    fun setUserContext(ctx: Long) {
+        this.userContext = ctx
```

Comment:
Can you explain the naming of `userContext`? If the idea based on C APIs which take a `void * userCtx` parameter, then I think the naming might be too generic for what it really is.

From my current understanding, I would just call this "pointerToSwiftAVPlayerInstance" or something equally specific. `userContext` in the C meaning of the word is something very generic that could be anything at all, which is why it's usually defined as `void*`. That doesn't really fit our use case here 🙏🏼